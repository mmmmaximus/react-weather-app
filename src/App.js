import React, {useState, useEffect} from 'react';
import axios from 'axios';
import Filter from './components/Filter'
import Countries from './components/Countries'

const App = () => {
  const [filter, setFilter] = useState('')
  const [countries, setCountries] = useState([])

  const changeFilter = (event) => {
    setFilter(event.target.value)
  }

  const showButton = (event) => {
    setFilter(event.target.value)
  }

  useEffect(() => {
    axios
      .get('https://restcountries.eu/rest/v2/all')
      .then(response => {
        setCountries(response.data)
      })
  }, [])

  return (
    <div>
      <Filter func={changeFilter}/>
      <Countries countries={countries} filter={filter} showButton={showButton}/>
    </div>
  )
}

export default App;
