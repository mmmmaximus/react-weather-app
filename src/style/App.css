* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  height: 100%;
  font-family: "Lucida Sans", sans-serif;
}

body {
  background-image: linear-gradient(to top, #2362bf, #007cd1, #0095de, #00ade8, #06c4ee);
  color: #ffffff;
}

.filter {
  padding: 20px 50px;
  text-align: center;
}

.countries {
  padding: 20px 50px;
}

.countries > .filter-text {
  text-align: center;
}

.countries > .countries-text {
  margin-left: 38%;
}

.country {
  display: flex;
  justify-content: space-between;
}

.country > div {
  width: 48%;
  padding: 10px;
}

.country-data > .img {
  width: 100%;
  max-width: 350px;
  margin-left: auto;
  margin-right: auto;
  display: block;
  padding: 20px;
}

.country-data > h2 {
  text-align: center;
  font-size: 1.5em;
  padding: 5px 0px 5px;
}

.country-data > table {
  margin: auto;
}

tr > td {
  padding: 5px 2px;
}

.languages > h3 {
  padding: 10px 0px;
  text-align: center;
}

div > li {
  padding: 0px 25px;
  margin-left: 35%;
}

.weather > .img {
  width: 85%;
  max-width: 300px;
  margin-left: auto;
  margin-right: auto;
  display: block;
  padding: 0px 20px;
}

.weather > h2 {
  text-align: center;
  font-size: 1.5em;
  padding: 0px 0px 5px 0px;
}

h3.description {
  text-align: center;
  padding: 5px 0px;
}

.weather > table {
  margin: auto;
}
