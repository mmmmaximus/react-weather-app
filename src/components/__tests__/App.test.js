import { render, screen } from '@testing-library/react'
import App from '../../App'
import renderer from 'react-test-renderer'
import mockedAxios from '../../__mocks__/axios'

test('App component renders', () => {
  const countriesResponse = { data: [] }
  mockedAxios.get.mockResolvedValueOnce(countriesResponse)
  render(<App />)
  const AppElement = screen.getByTestId('App')
  expect(AppElement).toBeInTheDocument
})

test('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
