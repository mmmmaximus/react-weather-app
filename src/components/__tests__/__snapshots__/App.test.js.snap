// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="App"
>
  <div
    className="filter"
  >
    find countries 
    <input
      data-testid="Filter"
      onChange={[Function]}
    />
  </div>
  <div
    className="countries"
    data-testid="Countries"
  >
    <p
      className="filter-text"
    >
      Search for a country
    </p>
  </div>
</div>
`;
